### PR TITLE
Recipe App setup

### PR DESCRIPTION
**Pull request by zamcham**
This pull request addresses the project requirements by implementing the necessary UI layout and functionality according to the provided wireframes. The following features have been implemented and tested thoroughly:

⚠️ 
**We have changed the name of the 'food' table to: Ingredients, and the 'food recipe' to 'ingredient ownership' to have better clarity while working**

1. **Login and Registration Pages (Devise):**
   - Created responsive login and registration pages using Devise.
   - Designed the pages with personalized colors, typography, and spacing.
   
2. **Food List (for groups with less than 3 members):**
   - Implemented a list of food items added by the logged-in user, displaying the quantity as required.
   - Added functionality to navigate to a form for adding new food items.
   
3. **Recipes List:**
   - Developed a list of recipes created by the logged-in user, adhering to the wireframe design.
   - Implemented functionality to access recipe details and delete owned recipes.
   
4. **Public Recipe List:**
   - Implemented a list of all public recipes, ordered by newest, matching the wireframe layout.
   - Users can navigate to recipe details from this list.
   
5. **Recipe Details:**
   - Added a toggle button allowing users to change the visibility of a recipe (public/private).
   - Displayed recipe details following the wireframe design, considering privacy and ownership.
   - Implemented functionality for recipe owners to add new food items to the recipe.
   
6. **General Shopping List (for groups with less than 3 members):**
   - Developed a view displaying missing food items for all recipes of the logged-in user.
   - Calculated the total count and price of the missing food items.
   - Optimized queries to prevent N+1 query issues.

7. **Navigation Menu:**
   - Created a navigation menu enabling users to access all the pages, enhancing user experience.
   
8. **Unit and Integration Tests:**
   - Implemented comprehensive unit and integration tests for all the developed features.
   - Ensured functionality, responsiveness, and edge cases were thoroughly tested.

**Notes:**
- This pull request fulfills the project requirements as specified.
- The UI design has been personalized while adhering to the wireframe structure.
- All features have been tested extensively to ensure proper functionality and responsiveness.
- The code has been reviewed for optimization and adherence to best practices.

Created by @naanmohammed  & @zamcham

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -6,11 +6,51 @@
...
+                <%= link_to "My Recipes", recipes_path, class: "nav-link" %>
+              </li>
+              <li class="nav-item">
+                <%= link_to "Ingredients", ingredients_path, class: "nav-link" %>
+              </li>
+              <li class="nav-item">
+                <%= link_to 'Shopping List', shopping_list_path, class: "nav-link" %>
+              </li>
+            <% end %>
+          </ul>
```
Comment (Owusu-Desmond) [inline app/views/layouts/application.html.erb:41]: - [ ] Please make sure you have a navigation link to public recipes page listing all publicly available recipes from other users with their names and total prices. The page has to match the structure of the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/public_recipe_index.png?raw=true) and url `/public_recipes`:pray: 
  
  image

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @zamcham and @naanmohammed , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the set up of the project ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,83 @@
+<div class="container hp">
+
```
Comment (addod19) [inline app/views/recipes/show.html.erb:2]: ### Optional
- [ ] It would be better if we add 2 buttons or links, that is 1. Generate shopping list and 2. Add Ingredients as we have in the wireframe. These are missing from our UI, kindly compare the images below
Screenshot from 2023-08-13 01-40-51
Screenshot from 2023-08-13 01-40-19

**Review by addod19 — APPROVED** [review]
## Hi @zamcham and @naanmohammed, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Awesome job addressing the comments from the previous review #:rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/show.html.erb (hunk 2 above)
Comment (naanmohammed) [reply app/views/recipes/show.html.erb:2]: Hello @addod19, Thank you very much for the quick review. The buttons are present when a user is signed in because a shopping list can only be generated by a signed in user and ingredients can only be added if the user is the owner of the recipe. 
Thank you very much again and have an awesome weekend.
